apis/Google.Devtools.AspNet/Google.Devtools.AspNet.IntegrationTests/CloudTrace/TraceTest.cs

```diff
@@ -0,0 +1,298 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using Google.Cloud.Trace.V1;
+using Google.Protobuf.Collections;
+using Google.Protobuf.WellKnownTypes;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+using Trace = Google.Cloud.Trace.V1.Trace;
+using TraceUtils = Google.Devtools.AspNet.Tests.TraceUtils;
+
+namespace Google.Devtools.AspNet.IntegrationTests
+{
+    public class TraceTest
+    {
+        private static readonly TraceIdFactory _traceIdFactory = TraceIdFactory.Create(); 
+
+        /// <summary>Total time to spend sleeping when looking for a trace.</summary>
+        private readonly TimeSpan _timeout = TimeSpan.FromSeconds(10);
+
+        /// <summary>Time to sleep between checks for a trace.</summary>
+        private readonly TimeSpan _sleepInterval = TimeSpan.FromSeconds(1);
+
+        /// <summary>Project id to run the test on.</summary>
+        private readonly string _projectId;
+
+        /// <summary>Unique id of the test.</summary>
+        private readonly string _testId;
+
+        /// <summary>Test start time to allow for easier querying of traces.</summary>
+        private readonly Timestamp _startTime;
+
+        /// <summary>Using the underlying grpc client so we can filter.</summary>
+        private readonly TraceService.TraceServiceClient _grpcClient;
+
+        public TraceTest()
+        {
+            _projectId = Utils.GetProjectIdFromEnvironment();
+            _testId = Utils.GetTestId();
+            _startTime = Timestamp.FromDateTime(DateTime.Now.ToUniversalTime());
+            _grpcClient = TraceServiceClient.Create().GrpcClient;
+        }
+
+        private GrpcTraceConsumer CreateGrpcTraceConsumer()
+        {
+            TraceServiceClient client = TraceServiceClient.Create();
+            return new GrpcTraceConsumer(Task.FromResult(client));
+        }
+
+        private SimpleManagedTracer CreateSimpleManagedTracer(ITraceConsumer consumer)
+        {
+            Trace trace = new Trace
+            {
+                ProjectId = _projectId,
+                TraceId = _traceIdFactory.NextId()
+            };
+            return SimpleManagedTracer.Create(consumer, trace, null);
+        }
+
+        /// <summary>
+        /// Block until the time has changed.
+        /// </summary>
+        private void BlockUntilClockTick()
+        {
+            DateTime now = DateTime.Now;
```

Comment:
Nit: Use UtcNow, always basically. In this particular case, we could have to wait a *long* time if someone runs this over the course of a DST transition... :)